src/type/definition.js

```diff
@@ -64,6 +72,14 @@ export type GraphQLInputType =
     GraphQLList<GraphQLInputType>
   >;
 
+export function assertInputType(type: ?GraphQLType): GraphQLInputType {
+  invariant(
+    type != null && isInputType(type),
```

Comment:
eqeqeq applies here.

These predicate functions accept `?Type` so you can fully lose the `type != null`